serving/docs/model.md

```diff
@@ -0,0 +1,384 @@
+
+public String processOutput(TranslatorContext ctx, NDList list) throws Exception;
+```
+
+### Provide Model File
+
+Next, you need to include a model file. DJL Serving supports model artifacts for the following engines:
+
+- MXNet
+- PyTorch (torchscript only)
+- TensorFlow
+- ONNX
+- PadddlePaddle
+
+You can also include a `synset.txt` file in the same directory with your model file to define the labels.
+
+### Packaging
+
+You can package the whole project as an uber jar by doing:
+
+```
+cd /path/to/model
+./gradlew jar
+```
+
+### Serving Example
+
+Let's run an example where we load a model in Java mode and run inference using the REST API.
+
+#### Step 1: Start Server
+
+First, start the server and load a model at startup. We will use a PyTorch resnet18 model as an example.
+
+```
+mkdir resnet18 && cd resnet18
+curl https://resources.djl.ai/test-models/traced_resnet18.pt -o resnet18.pt
+curl -O https://mlrepo.djl.ai/model/cv/image_classification/ai/djl/pytorch/synset.txt
+```
+
+Next, we need to prepare a Translator. In this example, we will use [CustomTranslator.java](https://github.com/deepjavalibrary/djl-demo/blob/master/djl-serving/java-mode/devEnv/src/main/java/CustomTranslator.java) in the djl-demo repo.
+
+```
+git clone https://github.com/deepjavalibrary/djl-demo.git
+cd djl-demo
+```
+
+The `Translator` is located at djl-serving/java-mode/devEnv. We need to copy the `Translator` to the classes folder.
+
+```
+mkdir -p resnet18/libs/classes
+cp djl-serving/java-mode/devEnv/src/main/java/CustomTranslator.java resnet18/libs/classes
+```
+
+Next, run the DJL Serving and load this model at startup.
+
+Linux/macOS
+
+```
+djl-serving -m resnet::PyTorch=file://$PWD/resnet18
+```
+
+Windows
+
+```
+path-to-your\serving.bat -m "resnet::PyTorch=file:///%cd%\resnet18"
+```
+
+This will launch the DJL Serving Model Server, bind to port 8080, and create an endpoint named `resnet` with the model.
+
+#### Step 2: Inference
+
+To query the model using the prediction API, open another session and run the following command:
+
+Linux/macOS
+
+```
+curl -O https://resources.djl.ai/images/kitten.jpg
+curl -X POST "http://127.0.0.1:8080/predictions/resnet" -T "kitten.jpg"
+```
+
+This should return the following result:
+
+```json
+[
+  {
+    "className": "n02124075 Egyptian cat",
+    "probability": 0.5183261632919312
+  },
+  {
+    "className": "n02123045 tabby, tabby cat",
+    "probability": 0.1956063210964203
+  },
+  {
+    "className": "n02123159 tiger cat",
+    "probability": 0.1955675184726715
+  },
+  {
+    "className": "n02123394 Persian cat",
+    "probability": 0.03224767744541168
+  },
+  {
+    "className": "n02127052 lynx, catamount",
+    "probability": 0.02553771249949932
+  }
+]
+```
+
+## Binary Mode
+
+This section walks through how to serve Binary model with DJL Serving.
+
+### Provide Model File
+
+For Binary Mode, you just need to place the model file in a folder.
```

Comment:
Binary mode doesn't support pre-processing and post-processing. DJLServing only accept Tensor (NDList/npy/npz) as input and output.

Reply:
OK